### PR TITLE
Remove deprecated provider block with special error handling

### PR DESCRIPTION
**Pull request by findkim**
`provider` block name was deprecated in [v0.1.0-techpreview2](https://github.com/hashicorp/consul-terraform-sync/blob/master/CHANGELOG.md#010-techpreview2-december-16-2020), it is removed in v0.1.0-beta.
 
image

related PR https://github.com/hashicorp/consul-terraform-sync/pull/140

### CODE REVIEW

**Review by lornasong — APPROVED** [review]
looks good to me!

File: config/config.go [hunk 1]
```diff
@@ -476,6 +463,26 @@ func supportedFormat(format string) bool {
 	return false
 }
 
+func processUnusedConfigKeys(md mapstructure.Metadata, file string) error {
```
Comment (eikenb) [inline config/config.go:466]: Erroring on unused configuration values is a great idea.

Comment (eikenb) [reply config/config.go:466]: Is there a test for this?

Comment (findkim) [reply config/config.go:466]: The config decoder had [already supported erroring unused configuration](https://github.com/hashicorp/consul-terraform-sync/pull/169/files#diff-fe44f09c4d5977b5f5eaea29170b6a0748819c9d02271746a20d81a5f3efca17R347
) and is [tested here](https://github.com/hashicorp/consul-terraform-sync/blob/master/config/config_test.go#L136-L149)

However with the way the decoder works, if that is set to `true`, it doesn't populate the metadata to figure out which keys are unused and returns an error. Another approach could string match that error for `'provider'` for this special case handling. So I swapped it to false and manually processed the unused keys.

**Review by eikenb — APPROVED** [review]

**Pull request merged**